src/config/apps.ts

```diff
@@ -49,7 +49,16 @@ import wavebox from './logos/wavebox.png'
 import yattee from './logos/yattee.png'
 import zoom from './logos/zoom.png'
 
-export const apps = {
+interface App {
+  name: string
+  logo: typeof firefox
+  privateArg?: string
+  convertUrl?: (url: string) => string
+}
+
+const typeApps = <T extends Record<string, App>>(apps: T) => apps
+
+export const apps = typeApps({
```

Comment:
what's the benefit of this over `as const` or simply 

```
export const apps: Record<string, App> = {
```

?

Reply:
It's the best of both worlds!

Compared to `const`:  auto completion and type checking when we edit the config:
<img width="760" alt="Screenshot 2022-03-04 at 11 41 47" src="https://user-images.githubusercontent.com/11351322/156748759-ba3187f2-fde5-4607-a1d7-b09ce604b052.png">
<img width="447" alt="Screenshot 2022-03-04 at 11 41 56" src="https://user-images.githubusercontent.com/11351322/156748770-bd9468fb-bfab-4b72-8030-e67a14543eef.png">

Compared to `const apps: Record<string, App>`: we still know exactly which browser is present and which browser has which property like in const before.
<img width="950" alt="Screenshot 2022-03-04 at 11 43 28" src="https://user-images.githubusercontent.com/11351322/156749195-5a0e357d-6694-4d31-b343-b5cc16dedf75.png">
<img width="275" alt="Screenshot 2022-03-04 at 11 43 58" src="https://user-images.githubusercontent.com/11351322/156749210-0dec7117-2c18-4e53-8efd-30a63cfef0d6.png">

Reply:
Nice! I'll be using this trick in the future for sure. Thanks.